libraries/freertos_plus/standard/freertos_plus_tcp/source/portable/NetworkInterface/Zynq/NetworkInterface.c

```diff
@@ -206,7 +221,27 @@ const TickType_t xWaitLinkDelay = pdMS_TO_TICKS( 7000UL ), xWaitRelinkDelay = pd
 
 BaseType_t xNetworkInterfaceOutput( NetworkBufferDescriptor_t * const pxBuffer, BaseType_t bReleaseAfterSend )
 {
-	if( ( ulPHYLinkStatus & BMSR_LINK_STATUS ) != 0 )
+	#if( ipconfigDRIVER_INCLUDED_TX_IP_CHECKSUM != 0 )
+	{
+	ProtocolPacket_t *pxPacket;
+
+		/* If the peripheral must calculate the checksum, it wants
+		the protocol checksum to have a value of zero. */
+		pxPacket = ( ProtocolPacket_t * ) ( pxBuffer->pucEthernetBuffer );
+		if( pxPacket->xICMPPacket.xIPHeader.ucProtocol == ipPROTOCOL_ICMP )
+		{
+		IPHeader_t *pxIPHeader = &( pxPacket->xUDPPacket.xIPHeader );
+
+			pxPacket->xICMPPacket.xICMPHeader.usChecksum = ( uint16_t )0u;
+			pxIPHeader->usHeaderChecksum = 0u;
+			pxIPHeader->usHeaderChecksum = usGenerateChecksum( 0UL, ( uint8_t * ) &( pxIPHeader->ucVersionHeaderLength ), ipSIZE_OF_IPv4_HEADER );
```

Comment:
The purpose of this block of code is that the #define above indicates that the driver itself should calculate the checksum, correct?

Reply:
The comments on lines 228/229 make it sound like the driver is using Zynq checksum offloading (CSO).  The code here is using 0 for the ICMP checksum at line 235 but generating an IP checksum at line 237 (after explicitly clearing it on line 236).  Wouldn't the CSO calculate the IP checksum?  I'm not that familiar with this feature, but [this Xilinx post](https://forums.xilinx.com/t5/Ethernet/How-to-enable-AXI-Ethernet-LWIP-Full-checksum-offload/m-p/895249) seems to indicate that you wouldn't need line 237 if CSO was enabled.  Please clarify the comments and adjust the code if needed.